apps/studio/components/interfaces/Connect/ConnectionPanel.tsx

```diff
@@ -0,0 +1,283 @@
+  lang?: CodeBlockLang
+  fileTitle?: string
+}
+
+const IPv4StatusIcon = ({ className, active }: { className?: string; active: boolean }) => {
+  return (
+    <div className={cn('relative inline-flex', className)}>
+      <svg
+        xmlns="http://www.w3.org/2000/svg"
+        fill="none"
+        viewBox="0 0 24 24"
+        stroke-width="1"
+        stroke="currentColor"
+        className="size-6 stroke-foreground-lighter"
+      >
+        <path
+          stroke-linecap="round"
+          stroke-linejoin="round"
+          d="M12 21a9.004 9.004 0 0 0 8.716-6.747M12 21a9.004 9.004 0 0 1-8.716-6.747M12 21c2.485 0 4.5-4.03 4.5-9S14.485 3 12 3m0 18c-2.485 0-4.5-4.03-4.5-9S9.515 3 12 3m0 0a8.997 8.997 0 0 1 7.843 4.582M12 3a8.997 8.997 0 0 0-7.843 4.582m15.686 0A11.953 11.953 0 0 1 12 10.5c-2.998 0-5.74-1.1-7.843-2.918m15.686 0A8.959 8.959 0 0 1 21 12c0 .778-.099 1.533-.284 2.253m0 0A17.919 17.919 0 0 1 12 16.5c-3.162 0-6.133-.815-8.716-2.247m0 0A9.015 9.015 0 0 1 3 12c0-1.605.42-3.113 1.157-4.418"
+        />
+      </svg>
+
+      {!active ? (
+        <div className="absolute -right-1.5 -top-1.5 bg-destructive rounded w-4 h-4 flex items-center justify-center">
+          <X size={10} strokeWidth={4} className="text-white rounded-full" />
+        </div>
+      ) : (
+        <div className="absolute -right-1.5 -top-1.5 bg-brand-500 rounded w-4 h-4 flex items-center justify-center">
+          <svg
+            width="10"
+            height="10"
+            viewBox="0 0 10 10"
+            fill="none"
+            xmlns="http://www.w3.org/2000/svg"
+          >
+            <path
+              d="M8.33325 2.5L3.74992 7.08333L1.66659 5"
+              stroke="white"
+              strokeWidth="2"
+              strokeLinecap="round"
+              strokeLinejoin="round"
+            />
+          </svg>
+        </div>
+      )}
+    </div>
+  )
+}
+
+export const CodeBlockFileHeader = ({ title }: { title: string }) => {
+  return (
+    <div className="flex items-center justify-between px-4 py-1 bg-surface-100/50 border border-b-0 border-surface rounded-t">
+      <div className="flex items-center gap-2">
+        <FileCode size={12} className="text-foreground-muted" strokeWidth={1.5} />
+        <span className="text-xs text-foreground-light">{title}</span>
+      </div>
+    </div>
+  )
+}
+
+export const ConnectionPanel = ({
+  type = 'direct',
+  title,
+  description,
+  connectionString,
+  onCopy,
+  ipv4Status,
+  notice,
+  parameters = [],
+  contentType = 'input',
+  lang = 'bash',
+  fileTitle,
+}: ConnectionPanelProps) => {
+  return (
+    <div className="flex flex-col gap-5 lg:grid lg:grid-cols-2 lg:gap-20 w-full">
+      <div className="flex flex-col">
+        <h1 className="text-sm mb-2">{title}</h1>
+        <p className="text-sm text-foreground-light mb-4">{description}</p>
+        <div className="flex flex-col -space-y-px">
+          {/* {contentType === 'input' ? ( */}
+          {false ? (
```

Comment:
hardcoded logic here

Reply:
will remove the component that's under `false` here - CodeBlock seems like the one that we want